### DISCUSSION TITLE
collection.getChildren(node.key) deprecation confusion

### DISCUSSION THREAD

**Discussion by binaryartifex** [adobe/react-spectrum#4348] [Q&A] [answered]
Hey folks, ive recently noticed a deprecation notice when trying to map over child nodes via

`Array.from(state.collection.body.childNodes).map((row)...`

I have to use Array.from against any react-aria collection otherwise strangely, my collections do not propagate in either storybook or my browser, both of which are the latest versions, but i digress. The suggestion is to use `collection.getChildren(node.key)` which is great until the realization that getChildren is an optional method, so i need to constantly force a non-null assertion against the method or check for the methods existence, which, doing it the latter way i need to either cater for an ultimate optional value or set a default empty set to avoid returning a possible null and still being able to run the map method. The current [table example](https://codesandbox.io/s/laughing-banzai-hx6105?file=/src/Table.tsx) also still does it the old way.

non null assertion
`Array.from(state.collection.getChildren!(headerRow.key)).map...`

the long way home
```typescript
      <thead {...theadGroupProps} className="sticky top-0" ref={headerRef}>
        {state.collection.headerRows.map((headerRow) => {
          const headerChildNodes =
            (state.collection.getChildren &&
              state.collection.getChildren(headerRow.key)) ||
            new Set();

          return (
            <TableHeaderRow key={headerRow.key} node={headerRow} state={state}>
              {Array.from(headerChildNodes).map((column) => {...
```

Would greatly appreciate some advice in regards to moving forward with iterating over child nodes in general, would be awesome to have a geez at some examples that use the non deprecated method too, cheers folks.

**Comment by devongovett** [comment, marked as answer]
Yeah we're in the middle of a transition to a new collection implementation so things are moving around a bit right now. We couldn't make `getChildren` required because that would be a breaking change for anyone implementing their own Collection. Maybe we shouldn't have added the deprecation comment just yet, not sure.

**Reply by binaryartifex** [reply]
ahh yep no worries cheers mate, ill sit tight for the moment then with the older implementation till you guys are done working your magic

**Reply by rolanday** [reply]
Please kindly update docs once sorted. E.g., https://react-spectrum.adobe.com/react-aria/useMenu.html shows deprecated usage. Thanks!! :-)

**Reply by BradRyan** [reply]
When trying to remove this deprecation warning on the Table, I am unable to get the following to work:
```{Array.from(collection.getChildren(collection.body.key)).map((row) => (```

I've reverted to using the following:
```{Array.from(collection.body.childNodes).map((row) => (```

If anyone know why, let me know please!
